### PR TITLE
RPC: SetPairingState open CommissioningWindow

### PR DESCRIPTION
**Pull request by rgoliver**
Problem:
Currently the RPC command SetPairingState only enables/disables Ble Advertising, it should also open a commissioning window.

Change:
If the commissioning window isn't already open, open a basic one when SetPairingState is true. Similarly close the commissioning window if SetPairingState is false.

Test:
Used RPC to toggle pairing off and back on and than successfully completed full commissioning with an m5.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: examples/common/pigweed/rpc_services/Device.h [hunk 1]
```diff
@@ -270,14 +270,46 @@ class Device : public pw_rpc::nanopb::Device::Service<Device>
...
     virtual pw::Status SetPairingState(const chip_rpc_PairingState & request, pw_protobuf_Empty & response)
     {
+        chip::ChipError err;
         if (request.pairing_enabled)
         {
-            DeviceLayer::ConnectivityMgr().SetBLEAdvertisingEnabled(true);
-            DeviceLayer::ConnectivityMgr().SetBLEAdvertisingMode(DeviceLayer::ConnectivityMgr().kFastAdvertising);
+            if (!DeviceLayer::ConnectivityMgr().IsBLEAdvertisingEnabled())
+            {
+                err = DeviceLayer::ConnectivityMgr().SetBLEAdvertisingEnabled(true);
```
Comment (bzbarsky-apple) [inline examples/common/pigweed/rpc_services/Device.h:278]: The comissioning window manager will do this already when the commissioning window is opened.  Why do we need to do it here?

File: examples/common/pigweed/rpc_services/Device.h [hunk 2]
```diff
@@ -270,14 +270,46 @@ class Device : public pw_rpc::nanopb::Device::Service<Device>
...
+                    return pw::Status::Internal();
+                }
+            }
         }
         else
         {
-            DeviceLayer::ConnectivityMgr().SetBLEAdvertisingEnabled(false);
+            if (DeviceLayer::ConnectivityMgr().IsBLEAdvertisingEnabled())
+            {
+                err = DeviceLayer::ConnectivityMgr().SetBLEAdvertisingEnabled(false);
```
Comment (bzbarsky-apple) [inline examples/common/pigweed/rpc_services/Device.h:301]: Again, CloseCommissioningWindow will do that....

File: examples/common/pigweed/rpc_services/Device.h (hunk 1 above)
Comment (rgoliver) [reply examples/common/pigweed/rpc_services/Device.h]: done

File: examples/common/pigweed/rpc_services/Device.h (hunk 2 above)
Comment (rgoliver) [reply examples/common/pigweed/rpc_services/Device.h]: done

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**